node/bft/src/primary.rs

```diff
@@ -2070,13 +2087,48 @@ mod tests {
 
         // The author must be known to resolver to pass propose checks.
         primary.gateway.resolver().insert_peer(peer_ip, peer_ip, peer_account.1.address());
+        // The primary must be considered synced.
+        primary.sync.block_sync().try_block_sync(&primary.gateway.clone()).await;
```

Comment:
NIT: Tests should be moved to a different file. Otherwise, it's unclear if this line of code was in a test. Commenting since I would have left a comment about the fact that we are taking a reference to a `clone`, i.e, the `clone` is pointless.

Reply:
I think that's a subjective decision which won't go out of this code anytime soon. Fortunately Github indicates `mod tests {` in the blue-grey summary line just above.